src/CTA.Rules.Update/WCF/WCFServicePort.cs

```diff
@@ -0,0 +1,330 @@
+                            SyntaxFactory.IdentifierName(netTcpMethodExpression)
+                            );
+                var netTcpInvocation = SyntaxFactory.InvocationExpression(netTCPExpression, SyntaxFactory.ArgumentList(SyntaxFactory.SeparatedList(new[] {
+                SyntaxFactory.Argument(SyntaxFactory.LiteralExpression(SyntaxKind.StringLiteralExpression, SyntaxFactory.Literal(transportPort.GetValueOrDefault(Constants.NettcpProtocol, 8000))))
+               })));
+                var kestrelInvocationWithNetTcp = kestrelInvocationNode.WithExpression(netTcpInvocation);
+                root = root.ReplaceNode(kestrelInvocationNode, kestrelInvocationWithNetTcp);
+            }
+
+            return root;
+        }
+
+        /// <summary>
+        /// Given existing Startup.cs template file, generate new Startup.cs based on configuration.
+        /// </summary>
+        /// <param name="startupFilePath">Path to existing Startup.cs template</param>
+        /// <returns>New Startup.cs contents</returns>
+        public string ReplaceStartupFile(string startupFilePath)
+        {
+            var startupFileContents = File.ReadAllText(startupFilePath);
+
+            if (_projectType == ProjectType.WCFConfigBasedService)
+            {
+                string configPath = Path.Combine(_projectPath, Constants.PortedConfigFileName);
+
+                if (HasBehavioursTag())
+                {
+                    startupFileContents = HandleBehaviorsTag(startupFilePath);
+                }
+
+                return startupFileContents.Replace(Constants.XMLPathPlaceholder, "@\"" + configPath + "\"");
+            }
+            else
+            {
+                string addService = Constants.AddServiceFormat;
+                string endpointConfigTemplate = Constants.AddServiceEndpointFormat;
+                string endpointConfigs = "";
+
+                ProjectWorkspace project = _analyzerResult.ProjectResult;
+
+                Dictionary<string, BindingConfiguration> bindingTransportMap = GetBindingsTransportMap(project);
+
+                Tuple<string, string> serviceInterfaceAndClass = WCFBindingAndTransportUtil.GetServiceInterfaceAndClass(project);
+
+                var serviceInterfaceName = serviceInterfaceAndClass.Item1 ?? Constants.DefaultServiceInterface;
+                var serviceClassName = serviceInterfaceAndClass.Item2 ?? Constants.DefaultServiceClass;
+
+                endpointConfigs += String.Format(addService, serviceClassName);
+
+                foreach(KeyValuePair<string, BindingConfiguration> keyValuePair in bindingTransportMap)
+                {
+                    var binding = keyValuePair.Key;
+                    var mode = keyValuePair.Value.Mode;
+                    var endpointAddress = keyValuePair.Value.EndpointAddress ?? String.Join("", "\\", "\"", binding.ToLower(), "\"");
+
+                    if (binding == Constants.HttpProtocol)
```

Comment:
if mode == None, just leave each block like new BasicHttpBinding() or new NetTcpBinding() or new WSHttpBinding(), otherwise do what you are doing... except the   binding == Constants.HttpsProtocol... for that I have written down below what it would be